### PR TITLE
CU-860rdgtu0 - separate metrics inventory plugin into deploy

### PR DESCRIPTION
**Pull request by Noy-Simon**
public: improve metric collection performance

### CODE REVIEW

File: charts/k8s-watcher/templates/metrics/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  {{- end }}
+  {{- if .Values.watcher.clusterName }}
+  - name: CLUSTER_NAME
+    value: {{ .Values.watcher.clusterName }}
+  {{- end }}
+{{- end }}
+{{- end }}
+
+{{- define "metrics.daemonset.volumes" }}
+{{- if and .Values.metrics (.Values.metrics.enabled | default false) }}
```
Comment (MickaelAlliel) [inline charts/k8s-watcher/templates/metrics/deploy.yaml:32]: wont this volume definition clash with the one the daemonset uses?

File: charts/k8s-watcher/templates/metrics/deploy.yaml [hunk 2]
```diff
@@ -28,7 +28,7 @@
 {{- end }}
 {{- end }}
 
-{{- define "metrics.daemonset.volumes" }}
+{{- define "metrics.deploy.volumes" }}
 {{- if and .Values.metrics (.Values.metrics.enabled | default false) }}
 - name: {{ include "k8s-watcher.name" . }}-metrics-config
```
Comment (MickaelAlliel) [inline charts/k8s-watcher/templates/metrics/deploy.yaml:33]: The volume name as well - it is the same of the daemonset

File: charts/k8s-watcher/templates/deployment.yaml [hunk 3]
```diff
@@ -41,9 +41,7 @@ spec:
         {{- include "metrics.init" . | indent 8 }}
       {{- end }}
       containers:
-        {{- if .Values.metrics.enabled }}
         {{- include "metrics.container" .        | indent 8 }}
```
Comment (MickaelAlliel) [inline charts/k8s-watcher/templates/deployment.yaml:44]: why did you remove the condition of if metrics is enabled?

File: charts/k8s-watcher/templates/metrics/deploy.yaml (hunk 2 above)
Comment (Noy-Simon) [reply charts/k8s-watcher/templates/metrics/deploy.yaml:33]: nope in the daemon its -daemon-config

File: charts/k8s-watcher/templates/deployment.yaml (hunk 3 above)
Comment (Noy-Simon) [reply charts/k8s-watcher/templates/deployment.yaml:44]: its already part of the metrics.container 
its redundant to do it twice

**Review by MickaelAlliel — APPROVED** [review]
